js-tasks/task1.js

```diff
@@ -0,0 +1,63 @@
+function summ() {
+  if (!Object.values(arguments).length) {
+    return null;
+  }
+
+  return Object.values(arguments).reduce(
+    (acc, next) =>
+      typeof next !== 'boolean' && !Number.isNaN(+next) ? acc + +next : acc,
+    0,
+  );
+}
+
+function summAdvanced() {
+  if (!Object.values(arguments).length) {
+    return null;
+  }
+
+  return Object.values(arguments).reduce((acc, next) => {
+    if (typeof next !== 'boolean' && !Number.isNaN(+next)) {
+      return acc + +next;
+    }
+
+    if (typeof next === 'function' && !Number.isNaN(+next())) {
+      return acc + +next();
+    }
+
+    return acc;
+  }, 0);
+}
+
+function isValueExists(a) {
+  if (a === null || typeof a === 'undefined') {
```

Comment:
typeof here is not really neccesary and is not usually used to check for undefined value
here is more informantion about difference
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined